hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCreate.java

```diff
@@ -191,6 +192,45 @@ public void testMkdirsByLevel() throws Exception {
     fs.mkdirs(new Path("a/b/c/d/e"));
   }
 
+  /*
+    Delete part of a path and validate subpath exists.
+   */
+  @Test
+  public void testMkdirsWithDelete() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    fs.mkdirs(new Path("a/b"));
+    fs.mkdirs(new Path("a/b/c/d"));
+    fs.delete(new Path("a/b/c/d"));
+    fs.getFileStatus(new Path("a/b/c"));
+    Assert.assertTrue(fs.exists(new Path("a/b/c")));
+  }
+
+  /**
+   * Verify mkdir and rename of parent.
+   */
+  @Test
+  public void testMkdirsWithRename() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    fs.mkdirs(new Path("a/b/c/d"));
+    fs.create(new Path("e/file"));
+    fs.delete(new Path("a/b/c/d"));
+    Assert.assertTrue(fs.rename(new Path("e"), new Path("a/b/c/d")));
+    Assert.assertTrue(fs.exists(new Path("a/b/c/d/file")));
+  }
+
+  /**
+   * Create a file with name /dir1 and then mkdirs for /dir1/dir2 should fail.
+   * @throws Exception
+   */
+  @Test
+  public void testFileCreateMkdirsRoot() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    fs.setWorkingDirectory(new Path("/"));
+    final Path p1 = new Path("dir1");
+    fs.create(p1);
+    intercept(IOException.class, () -> fs.mkdirs(new Path("dir1/dir2")));
+  }
+
   /**
```

Comment:
Is there a way we can add tests for the scenarios we discussed on the chat??
Like:
/user/snvijaya/a/b/c all implicit
marker for b
mkdir(a/b/c/d)

Assert user, snvijaya and a are implicit
Assert b, c, d are explicit.

Reply:
Added this test